packages/davinci-client/src/lib/client.store.ts

```diff
@@ -110,5 +74,80 @@ export async function davinci({ config }: { config: DaVinciConfig }) {
         store.dispatch(nodeSlice.actions.update({ id, value, index }));
       };
     },
+
+    /**
+     * @method client - Selector to get the node.client from state
+     * @returns {Node.client} - the client property from the current node
+     */
+    getClient: () => nodeSlice.selectors.selectClient(store.getState()),
+
+    /**
+     * @method collectors - Selector to get the collectors from state
+     * @returns {Collector[]} - The collectors from the current node in state
+     */
+    getCollectors: () => {
+      const state = store.getState();
+      const client = nodeSlice.selectors.selectClient(state);
+      // Let's check if the node has a client and collectors
+      if (client && 'collectors' in client) {
+        return nodeSlice.selectors.selectCollectors(state) || [];
+      }
+      // Return an empty array if no client or collectors are found
+      return [];
+    },
+
+    getError: () => {
+      const state = store.getState();
+      return nodeSlice.selectors.selectError(state);
+    },
+
+    /**
+     * @method node - Selector to get the node from state
+     * @returns {Node} - the current node from state
+     */
+    getNode: () => {
+      return nodeSlice.selectSlice(store.getState());
+    },
+
+    /**
+     * @method server - Selector to get the node.server from state
+     * @returns {Node.server} - the server property from the current node
+     */
+    getServer: () => {
+      const state = store.getState();
+      return nodeSlice.selectors.selectServer(state);
+    },
+
+    /**
+     * Utilities to help query cached responses from server
+     */
+    cache: {
+      getLatestResponse: () => {
+        const node = nodeSlice.selectSlice(store.getState());
+
+        if (!node.cache?.key) {
+          console.error(`Cannot find current node's cache key or no current node`);
+          return null;
+        }
+
+        const flowItem = davinciApi.endpoints.flow.select(node.cache.key);
+        const nextItem = davinciApi.endpoints.next.select(node.cache.key);
+        const startItem = davinciApi.endpoints.start.select(node.cache.key);
+
+        return flowItem || nextItem || startItem;
+      },
+      getResponseWithId: (requestId: string) => {
+        if (!requestId) {
+          console.error('Please provide the cache key');
+          return null;
+        }
+
+        const flowItem = davinciApi.endpoints.flow.select(requestId);
+        const nextItem = davinciApi.endpoints.next.select(requestId);
+        const startItem = davinciApi.endpoints.start.select(requestId);
+
+        return flowItem || nextItem || startItem;
+      },
+    },
```

Comment:
Selectors specifically to access cached responses from DaVinci. This is a more advanced method for allowing developers to access highly specific properties from the response that isn't a part of the normalized "node" state.